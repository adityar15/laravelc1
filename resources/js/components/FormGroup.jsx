import React from "react"

export default function FormGroup({
    label,
    name = "",
    type = "text",
    className = "",
    placeholder = "",
    errorMessage = "",
    ...props
}) {
    return (
        <div className={`flex flex-col gap-2 ${className}`}>
            <label className='text-lg font-semibold text-gray-800'>
                {label}
            </label>
            
            {type == "textarea" ?  <textarea name={name} placeholder={placeholder} type={type} className="border border-gray-600 rounded-md p-2" {...props}></textarea> :  
            <input name={name} placeholder={placeholder} type={type} className="border border-gray-600 rounded-md p-2" {...props} />
            }
            
          
            
            
            
            {errorMessage && <span className="text-red-500 text-sm">{errorMessage}</span>}
        </div>
    )
}

