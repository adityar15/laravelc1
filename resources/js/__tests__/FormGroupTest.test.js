import React from "react"
import renderer from 'react-test-renderer';

import FormGroup from "../components/FormGroup";

test('renders the FormGroup with props', () => {
  
  const props = {
    label: "Test Label",
    name: "testName",
    type: "text",
    className: "test-class",
    placeholder: "Enter text",
    errorMessage: "Test error message"
  };

    // match snapshot of entire component
    const component = renderer.create(<FormGroup {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();


  });