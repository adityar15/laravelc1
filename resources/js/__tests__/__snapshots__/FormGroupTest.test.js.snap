// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the FormGroup with props 1`] = `
<div
  className="flex flex-col gap-2 test-class"
>
  <label
    className="text-lg font-semibold text-gray-800"
  >
    Test Label
  </label>
  <input
    className="border border-gray-600 rounded-md p-2"
    name="testName"
    placeholder="Enter text"
    type="text"
  />
  <span
    className="text-red-500 text-sm"
  >
    Test error message
  </span>
</div>
`;
